tests/cleantest.py

```diff
@@ -0,0 +1,160 @@
+"""
+cleantest enables unittest test cases and suites to be run in separate python
+interpreter instances, in parallel.
+"""
+from collections import namedtuple
+import multiprocessing.dummy
+import pickle
+import unittest
+import subprocess
+
+
+DEFAULT_NUM_PROCS = 8
+
+
+def cleantest(obj):
+    """
+    Marks a test case that is to be run in its own 'clean' interpreter instance.
+
+    When applied to a TestCase class, each method will be run in a separate
+    interpreter instance, in parallel.
+
+    Usage on a class::
+
+        @clean
+        class PatchTests(object):
+            # will be run in new interpreter
+            def test_patch_before_import(self):
+                patch()
+                import module
+
+            # will be run in new interpreter as well
+            def test_patch_after_import(self):
+                import module
+                patch()
+
+
+    Usage on a test method::
+
+        class OtherTests(object):
+            @clean
+            def test_case(self):
+                pass
+
+
+    :param obj: method or class to run in a separate python interpreter.
+    :return:
+    """
+    setattr(obj, '_test_clean', True)
+    return obj
+
+
+def is_iterable(i):
+    try:
+        iter(i)
+    except TypeError:
+        return False
+    else:
+        return True
+
+
+def is_cleantest(test):
+    if hasattr(test, '_test_clean'):
+        return True
+    if hasattr(test, '_testMethodName'):
+        t = getattr(test, test._testMethodName)
+        if hasattr(t, '_test_clean'):
+            return True
+    return False
+
+
+class CleanTestSuite(unittest.TestSuite):
+    TestResult = namedtuple('TestResult', 'test returncode output')
+
+    def __init__(self, modprefix, *args, **kwargs):
+        self.modprefix = modprefix
+        self.num_procs = kwargs.get('num_procs', DEFAULT_NUM_PROCS)
+        super(CleanTestSuite, self).__init__(*args, **kwargs)
+
+    @staticmethod
+    def merge_result(into_result, new_result):
+        into_result.failures += new_result.failures
+        into_result.errors += new_result.errors
+        into_result.skipped += new_result.skipped
+        into_result.expectedFailures += new_result.expectedFailures
+        into_result.unexpectedSuccesses += new_result.unexpectedSuccesses
+        into_result.testsRun += new_result.testsRun
+
+    @staticmethod
+    def get_tests_from_suite(suite):
+        tests = []
+        clean_tests = []
+        suites_to_check = [suite]
+        while suites_to_check:
+            suite = suites_to_check.pop()
+            for t in suite:
+                if is_iterable(t):
```

Comment:
```suggestion
                if isinstance(t, collections.Iterable):
```

Reply:
Darn, I thought I made this change already from the comment on the other PR. 😕